tutorial/numpy-functions.md

```diff
@@ -0,0 +1,350 @@
+    prd = product(arr)
+    print *, 'prd: ' // ' ' , prd
+
+    end program prog_test_prod
+    ```
+
+## [mod](https://numpy.org/doc/stable/reference/generated/numpy.mod.html)
+
+-   Supported parameters:
+
+    x1: array_like
+        Dividend array.
+
+    x2: array_like,
+        Divisor array. If x1.shape != x2.shape, they must be
+        broadcastable to a common shape (which becomes the shape of the output).
+
+-   Supported language: fortran.
+
+-   python code:
+
+    ```python
+    from numpy import array, mod
+
+    arr = array([1,2,3,4])
+    res = mod(arr, arr)
+    print("res: ", res)
+    ```
+
+-   fortran equivalent:
+
+    ```fortran
+    program prog_test_mod
+
+    use, intrinsic :: ISO_C_BINDING
+
+    implicit none
+
+    integer(C_INT64_T), allocatable :: arr(:)
+    integer(C_INT64_T), allocatable :: res(:)
+
+    allocate(arr(0:4_C_INT64_T - 1_C_INT64_T))
+    arr = [1_C_INT64_T, 2_C_INT64_T, 3_C_INT64_T, 4_C_INT64_T]
+    allocate(res(0:4_C_INT64_T - 1_C_INT64_T))
+    res = modulo(arr,arr)
+    print *, 'res: ' // ' ' , res
+
+    end program prog_test_mod
+    ```
+
+## [matmul](https://numpy.org/doc/stable/reference/generated/numpy.matmul.html)
+
+-   Supported parameters:
+
+    x1, x2: array_like,
+        Input arrays, scalars not allowed.
+
+-   Supported langauges: fortran
+
+-   python code:
+
+    ```python
+    from numpy import array, matmul
+
+    arr = array([[1,2],[3,4]])
+    res = matmul(arr, arr)
+    print("res: ", res)
+    ```
+
+-   fortran equivalent:
+
+    ```fortran
+    program prog_test_matmul
+
+    use, intrinsic :: ISO_C_BINDING
+
+    implicit none
+
+    integer(C_INT64_T), allocatable :: arr(:,:)
+    integer(C_INT64_T), allocatable :: res(:,:)
+
+    allocate(arr(0:2_C_INT64_T - 1_C_INT64_T, 0:2_C_INT64_T - 1_C_INT64_T))
+    arr = reshape([[1_C_INT64_T, 2_C_INT64_T], [3_C_INT64_T, 4_C_INT64_T]], &
+          [2_C_INT64_T, 2_C_INT64_T])
+    allocate(res(0:2_C_INT64_T - 1_C_INT64_T, 0:2_C_INT64_T - 1_C_INT64_T))
+    res = matmul(arr,arr)
+    print *, 'res: ' // ' ' , res
+
+    end program prog_test_matmul
+    ```
+
+## [Numpy types](https://numpy.org/devdocs/user/basics.types.html)
+
+-   Supported types : bool, int, int8, int16, int32, int64, float, float32, float64, complex64 and complex128. they can be used as cast functions too.
+
+## Other functions
+
+-   Supported [math functions](https://numpy.org/doc/stable/reference/routines.math.html):
+
+    NumpySqrt, NumpyAbs, NumpySin, NumpyCos, NumpyExp, NumpyLog, NumpyTan, NumpyArcsin, NumpyArccos, NumpyArctan, NumpyArctan2, NumpySinh, NumpyCosh, NumpyTanh, NumpyArcsinh, NumpyArccosh and
+    NumpyArctanh.
+
+-   Supported arrays manipulation:
+
+    NumpyEmpty, NumpyEmptyLike, NumpyFull, NumpyFullLike, NumpyOnes,
+    NumpyOnesLike, NumpyZeros and NumpyZerosLike.
+
+-   others:
+
+    NumpyMax, NumpyMin, NumpySum, NumpyRand, NumpyRandint, Shape, NumpyArange, NumpyFloor.
```

Comment:
Are all arguments supported here? If not we should specify which ones are/aren't

Reply:
we have different max, min, and floor from the numpy ones, for the others most of them are just missing `like` parameter.

Reply:
What is the difference for the max/min/floor functions?
> for the others most of them are just missing `like` parameter.

Please add this to the documentation

Reply:
our max/min functin are confusing somehow, they take one argument `x`, for numpy they two functions each, numpy.ndarray.max / numpy.maximum and numpy.ndarray.mini / numpy.minimum. For floor we just don't support the other opsional arguments.

Reply:
I suspect that we support `numpy.ndarray.max` rather than `numpy.maximum` (see [amax](https://numpy.org/doc/stable/reference/generated/numpy.amax.html)) (same for min)